R/imputation.R

```diff
@@ -26,7 +26,11 @@ setMethod("impute", "MSnSet",
               ## imputaton methods
               ##
               if (method == "knn") {
-                  .eset <- impute.knn(exprs(object), ...)$data
+                  imp_res <- impute.knn(exprs(object), ...)
+                  .eset <- imp_res$data
+                  if(! is.null(imp_res$rng.state)){
+                    .Random.seed <<- imp_res$rng.state
+                  }
```

Comment:
I am not sure but what about a `else` statement. If `.Random.seed` wasn't set at all. It should not be created here, or?
```suggestion
                  } else {
                      rm(".Random.seed", envir=.GlobalEnv)
                  }
```

Reply:
Yes you are right, I think that is probably the best thing one can do in that case.